### ISSUE TITLE
Brian configuration summary

### ISSUE THREAD

**Issue by thesamovar** [brian-team/brian2#641]
New function ``brian_configuration()`` that returns a string summarising current configuration options.

**Comment by Dyex719** [comment]
Should I edit this file by adding a new function brian_configuration() https://github.com/brian-team/brian2/blob/9243dac858d766f22423f8855dc22fded627f6f5/docs_sphinx/conf.py ?

**Comment by mstimberg** [comment]
> Should I edit this file by adding a new function brian_configuration() https://github.com/brian-team/brian2/blob/9243dac858d766f22423f8855dc22fded627f6f5/docs_sphinx/conf.py ?

That file is the configuration file for the documentation system. The idea about `brian_configuration` was rather to have it return configuration options in the sense of the preferences (e.g. code generation target). Or did I misunderstand @thesamovar ? Either way, we should add a few details to this issue report.

**Comment by thesamovar** [comment]
Yeah the idea was that it should return something a bit like the autogenerated preference list in the documentation listing the current preferences. Could also maybe have an option to list only the non-default valued preferences.

**Comment by sukarxn** [comment]
please check https://github.com/brian-team/brian2/pull/1806
